api/wasm/v0/cpp/proxy_wasm_externs.h

```diff
@@ -0,0 +1,151 @@
+/*
+ * Proxy-WASM ABI.
+ */
+// NOLINT(namespace-envoy)
+
+#pragma once
+
+#include <stddef.h>
+#include <stdint.h>
+
+//
+// ABI calls from the host into the VM.
+//
+// These will typically be implemented by a language specific SDK which will provide an API on top
+// of this ABI e.g. the C++ SDK provides a proxy_wasm_api.h implementation of the API on top of
+// this ABI.
+//
+// The Wasm VM can only access memory in the VM. Consequently, all data must be passed as integral
+// call parameters or by the host allocating memory in the VM which is then owned by the Wasm code.
+// For consistency and to enable diverse Wasm languages (e.g. languages with GC), the ABI uses a
+// single mechanism for allocating memory in the VM and requires that all memory allocations be
+// explicitly requested by calls from the VM and that the Wasm code then owns the allocated memory.
+//
+
+// Non-stream calls.
+
+/**
+ * Called when the VM starts by the first plugin to use the VM.
+ * @param root_context_id is an identifier for one or more related plugins.
+ * @param vm_configuration_size is the size of any configuration available via
+ * proxy_get_configuration during the lifetime of this call.
+ * @return non-zero on success and zero on failure (e.g. bad configuration).
+ */
+enum class OnVmStartResult : uint32_t {
+  BadConfiguration = 0,
+  Ok = 1,
```

Comment:
We're not reusing `WasmResult` here? I was thinking that was supposed to be generic.

Reply:
This is also a bit confusing, as `WasmResult` is 0 for OK. Prefer to have consistent positive/negative logic.

Reply:
I see, `WasmResult` is for VM --> host, this is the other way around. Still, prefer consistent positive/negative logic.

Reply:
When it was a boolean before it made sense that 1 was success, but now that it is an enum, I agree.  Changed so that we have 0 as success.  PTAL